### PR TITLE
fix: specific check for use foreman in github codespaces

### PR DESCRIPTION
**Pull request by tanmoysrt**
#### Fix
In github codespaces, it is running the configuration for docker environment but it should start the processes using foreman.

#### Describe the changes you have made in this PR -
- In `/bin/dev` checks added for `REMOTE_DEVELOPMENT_PLATFORM` env variable. If it's set it will not run the command for docker one . Instead it will start all processes using foreman

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: .devcontainer/docker-compose.yml [hunk 1]
```diff
@@ -17,7 +17,8 @@ services:
     environment:
       REDIS_URL: "redis://redis:6379/0"
       CIRCUITVERSE_USE_SOLR: "false"
-      DOCKER_ENVIRONMENT: 'true'
+      DOCKER_ENVIRONMENT: "true"
+      REMOTE_DEVELOPMENT_PLATFORM: "true"
```
Comment (tachyons) [inline .devcontainer/docker-compose.yml:21]: dev containers are used for local development as well right ?

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -3,7 +3,7 @@ FROM ruby:3.2.1
 # install dependencies
 RUN apt-get update -qq && apt-get install -y imagemagick shared-mime-info libvips && apt-get clean
 
-RUN curl -sL https://deb.nodesource.com/setup_14.x | bash \
+RUN curl -sL https://deb.nodesource.com/setup_16.x | bash \
```
Comment (tachyons) [inline .devcontainer/Dockerfile:6]: Could you use move node update to separate PR ?

Comment (tanmoysrt) [reply .devcontainer/Dockerfile:6]: Ok :+1:

File: .devcontainer/docker-compose.yml (hunk 1 above)
Comment (tanmoysrt) [reply .devcontainer/docker-compose.yml:21]: For locally, the recommended method was to use the docker environment by `bin/docker_run`. 

But anyone can use the provided configuration to start the dev container in vscode. We haven't tested or documented this. This configuration was specifically added for Github Codespaces

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (tanmoysrt) [reply .devcontainer/Dockerfile]: updated !

File: .devcontainer/docker-compose.yml (hunk 1 above)
Comment (tachyons) [reply .devcontainer/docker-compose.yml:21]: @tanmoysrt then Can we us the env DEV_CONTAINER to make this more specific ?

Comment (tanmoysrt) [reply .devcontainer/docker-compose.yml]: @tachyons as there is no pre-existing environment variable like DEV_CONTAINER in the codespaces environment. Added the DEV_CONTAINER variable in docker-compose and updated
